service/controller/release/resource/status/create.go

```diff
@@ -0,0 +1,73 @@
+package status
+
+import (
+	"context"
+	"fmt"
+
+	appv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/release-operator/pkg/project"
+	"github.com/giantswarm/release-operator/service/controller/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	release, err := key.ToReleaseCR(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if release.DeletionTimestamp != nil {
+		return nil
+	}
+
+	components := key.FilterComponents(release.Spec.Components)
+
+	var apps appv1alpha1.AppList
+	{
+		err := r.k8sClient.CtrlClient().List(
+			ctx,
+			&apps,
+			&client.ListOptions{
+				LabelSelector: labels.SelectorFromSet(labels.Set{
+					key.LabelManagedBy: project.Name(),
+				}),
+			},
+		)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	var releaseDeployed bool
+	{
+		releaseDeployed = true
+		for _, component := range components {
+			if !key.ComponentDeployed(component, apps.Items) {
+				releaseDeployed = false
+
+				r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("component %#q in release %#q is not deployed", component.Name, release.Name))
+			}
+		}
+	}
+
+	{
+		// TODO: Actually check if we actually need to update
```

Comment:
Not a big deal to either check or remove TODO here

Reply:
Removed here, we will have a follow up story for some improvements.